aws/elasticsearch_domain_structure.go

```diff
@@ -48,10 +49,24 @@ func flattenAdvancedSecurityOptions(advancedSecurityOptions *elasticsearch.Advan
 		return []map[string]interface{}{}
 	}
 
-	m := map[string]interface{}{
-		"enabled":                        aws.BoolValue(advancedSecurityOptions.Enabled),
-		"internal_user_database_enabled": aws.BoolValue(advancedSecurityOptions.InternalUserDatabaseEnabled),
+	m := map[string]interface{}{}
+	if aws.BoolValue(advancedSecurityOptions.Enabled) {
+		m["enabled"] = aws.BoolValue(advancedSecurityOptions.Enabled)
+		m["internal_user_database_enabled"] = aws.BoolValue(advancedSecurityOptions.InternalUserDatabaseEnabled)
 	}
```

Comment:
To handle the perpetual diff in cases where `advancedSecurityOptions.InternalUserDatabaseEnabled` is set to `true` but `advancedSecurityOptions.Enabled` is set to `false` in a config so the API returns `false` for both of these values